File: sistema-cines/src/models/Counts/role.ts
export interface Role {
  idRol: number;
  nombreRol: string;
  descripcion: string;
}

//getters 
export function getIdRol(role: Role): number {
  return role.idRol;
}

export function getNombreRol(role: Role): string {
  return role.nombreRol;
}

export function getDescripcion(role: Role): string {
  return role.descripcion;
}

//setters
export function setIdRol(role: Role, idRol: number): void {
  role.idRol = idRol;
}

export function setNombreRol(role: Role, nombreRol: string): void {
  role.nombreRol = nombreRol;
}

export function setDescripcion(role: Role, descripcion: string): void {
  role.descripcion = descripcion;
}

//constructor
export function createRole(idRol: number, nombreRol: string, descripcion: string): Role {
  return {
    idRol,
    nombreRol,
    descripcion
  };
}

File: rest-api-ipc2/src/main/java/models/users/Role.java
package models.users;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Role {
    private int idRol;
    private String nombreRol;
    private String descripcion;

    // Constructor por defecto necesario para Jackson
    public Role() {
    }

    // Constructor con parámetros
    @JsonCreator
    public Role(@JsonProperty("idRol") int idRol, 
                @JsonProperty("nombreRol") String nombreRol, 
                @JsonProperty("descripcion") String descripcion) {
        this.idRol = idRol;
        this.nombreRol = nombreRol;
        this.descripcion = descripcion;
    }

    // Getters y setters
    public int getIdRol() {
        return idRol;
    }

    public void setIdRol(int idRol) {
        this.idRol = idRol;
    }

    public String getNombreRol() {
        return nombreRol;
    }

    public void setNombreRol(String nombreRol) {
        this.nombreRol = nombreRol;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
}